vm/vm.go

```diff
@@ -353,6 +379,20 @@ func (vm *VM) Initialize(
 	vm.syncer = validitywindow.NewSyncer(vm, vm.chainTimeValidityWindow, func(time int64) int64 {
 		return vm.ruleFactory.GetRules(time).GetValidityWindow()
 	})
+	vm.acceptedSubscriptions = append(vm.acceptedSubscriptions, event.SubscriptionFunc[*StatefulBlock]{
+		AcceptF: func(b *StatefulBlock) error {
+			seenValidityWindow, err := vm.syncer.Accept(context.TODO(), b.ExecutionBlock)
+			if err != nil {
+				vm.Fatal("syncer failed to accept block", zap.Error(err))
+			}
+			if seenValidityWindow {
+				vm.seenValidityWindowOnce.Do(func() {
+					close(vm.seenValidityWindow)
+				})
+			}
+			return nil
```

Comment:
Rather than check if the `seenValidityWindow` channel is closed, always mark the block as accepted, which in turn pushes it to the time validity window and rely on `seenValidityWindowOnce` to make sure we only close the `seenValidityWindow` channel once.